### PR TITLE
convert ALL the fmt.Errorf to errors.New if there is no need to format the string

### PR DESCRIPTION
**Pull request by kevinetc123**
as the Go source said , `fmt.Errorf` is just the combination of `fmt.Sprintf` and `errors.New`:
```go
// Errorf formats according to a format specifier and returns the string
// as a value that satisfies error.
func Errorf(format string, a ...interface{}) error {
	return errors.New(Sprintf(format, a...))
}
```
Signed-off-by: kaiwentan <kaiwentan@harmonycloud.cn>

### CODE REVIEW

File: manifest/schema1/reference_builder.go [hunk 1]
```diff
@@ -58,7 +57,7 @@ func (mb *referenceManifestBuilder) Build(ctx context.Context) (distribution.Man
 func (mb *referenceManifestBuilder) AppendReference(d distribution.Describable) error {
 	r, ok := d.(Reference)
 	if !ok {
-		return fmt.Errorf("Unable to add non-reference type to v1 builder")
+		return errors.New("Unable to add non-reference type to v1 builder")
```
Comment (dmcgowan) [inline manifest/schema1/reference_builder.go]: Since you touched this line can you make "Unable" lower case

**Comment by allencloud** [comment]
I think the design is good. However I think we should change all the improper things in a single time.
Could you use grep to find all these things and change them all in this PR?

File: configuration/configuration.go [hunk 2]
```diff
@@ -627,7 +628,7 @@ func Parse(rd io.Reader) (*Configuration, error) {
 						v0_1.Loglevel = Loglevel("info")
 					}
 					if v0_1.Storage.Type() == "" {
-						return nil, fmt.Errorf("No storage configuration provided")
+						return nil, errors.New("No storage configuration provided")
```
Comment (stevvooe) [inline configuration/configuration.go]: Lower case here, as well.

Comment (kevinetc123) [inline configuration/configuration.go]: ok

File: manifest/schema1/reference_builder.go (hunk 1 above)
Comment (kevinetc123) [inline manifest/schema1/reference_builder.go]: @dmcgowan  ok, the PR has been updated

**Comment by kevinetc123** [comment]
@allencloud  good idea , I have updated the whole repo by global regex replace .

**Comment by allencloud** [comment]
I am afraid we need a rebase to solve the conflict.

**Comment by kevinetc123** [comment]
@allencloud  done!

**Comment by stevvooe** [comment]
Why?

To be honest, I use `fmt.Errorf` when treating the error as a specific string.

**Comment by dmcgowan** [comment]
I would like to see this change be part of a larger move to `github.com/pkg/errors`. This would also entail updating the imports and using wrap. Just changing `fmt.Errorf` to `errors.New` is not providing any benefit to the code, however switching to `github.com/pkg/errors` does provide a clear benefit and would also entail switching all `fmt.Errorf` to `errors.Errorf` and wrap when providing useful contextual information.

**Comment by dmcgowan** [comment]
Closing as non-functional change which does not make improvement to code quality. If you want to invest in better error handling and messages feel free to open a new PR.

**Pull request closed without merging**